src/appMain/CMakeLists.txt

```diff
@@ -251,3 +257,7 @@ else()
       GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 endif()
 
+if(BUILD_TESTS)
+  add_library("LowVoltageHandlerLibrary" ${CMAKE_CURRENT_SOURCE_DIR}/low_voltage_signals_handler.cc)
```

Comment:
@AByzhynar In case BUILD_TESTS option is set file low_voltage_signals_handler.cc will be compiled two times.
To avoid this u can:
 - create library AppService that includes
   ${CMAKE_CURRENT_SOURCE_DIR}/life_cycle_impl.cc - that is also may be used in tests
   ${CMAKE_CURRENT_SOURCE_DIR}/low_voltage_signals_handler.cc
   ${CMAKE_CURRENT_SOURCE_DIR}/signal_handlers.cc - that is also may be used in tests
then link it to SmartDeviceLinkCore and to test
2) create OBJECT library 
https://cmake.org/cmake/help/v3.0/command/add_library.html
then link it in appropriate way to you test and executable

Reply:
@AByzhynar Also I suppose the library type should be specified manually.
Because if BUILD_SHARED is set then install rules should be provided.
Also build broken for now if BUILD_SHARED is set to ON.

Reply:
@EKuliiev What is the library type `OBJECT` ? I don't know such type. Can you please explain or provide a link?

Reply:
@AByzhynar see https://cmake.org/cmake/help/v3.9/command/add_library.html#object-libraries
# Create obj lib
add_library("LowVoltageHandlerLibrary" OBJECT ${CMAKE_CURRENT_SOURCE_DIR}/low_voltage_signals_handler.cc
)
....
set (SOURCES
  ${CMAKE_CURRENT_SOURCE_DIR}/main.cc
  ${CMAKE_CURRENT_SOURCE_DIR}/life_cycle_impl.cc
  ${CMAKE_CURRENT_SOURCE_DIR}/signal_handlers.cc
$<TARGET_OBJECTS:LowVoltageHandlerLibrary>
)

...
set(testSources
  ${CMAKE_SOURCE_DIR}/src/appMain/test/low_voltage_signals_handler_test.cc
  $<TARGET_OBJECTS:LowVoltageHandlerLibrary>
)

Reply:
Such problem exists in test_main.cc
if u change it, test_main.ccc will be complied number of unittests it used times

Reply:
@EKuliiev Thank you.

Reply:
@EKuliiev fixed in 595cee4db06641f9fc015157dc8f879958c299d2